src/boot/efi/stub.c

```diff
@@ -159,6 +163,18 @@ EFI_STATUS efi_main(EFI_HANDLE image, EFI_SYSTEM_TABLE *sys_table) {
         if (err != EFI_SUCCESS)
                 return log_error_status_stall(err, L"Error getting a LoadedImageProtocol handle: %r", err);
 
+        if (efivar_get_uint64_le(LOADER_GUID, L"LoaderFeatures", &loader_features) != EFI_SUCCESS ||
+            !FLAGS_SET(loader_features, EFI_LOADER_FEATURE_RANDOM_SEED)) {
+                _cleanup_(file_closep) EFI_FILE *esp_dir = NULL;
+                EFI_HANDLE esp_device;
+
+                err = partition_open(ESP_GUID, loaded_image->DeviceHandle, &esp_device, &esp_dir);
+                if (err != EFI_SUCCESS)
+                        return log_error_status_stall(err, L"Unable to find ESP directory: %r", err);
```

Comment:
@poettering Should this be non-fatal, for cases where UKI is booted from some weird scenario?